src/Microsoft.NetCore.Analyzers/Core/Security/DoNotAddArchiveItemPathToTheTargetFileSystemPath.cs

```diff
@@ -0,0 +1,141 @@
+                                OperationKind.PropertyReference);
+
+                            operationBlockStartContext.RegisterOperationBlockEndAction(
+                                operationBlockAnalysisContext =>
+                                {
+                                    if (!rootOperationsNeedingAnalysis.Any())
+                                    {
+                                        return;
+                                    }
+
+                                    foreach (IOperation rootOperation in rootOperationsNeedingAnalysis)
+                                    {
+                                        var taintedDataAnalysisResult = TaintedDataAnalysis.TryGetOrComputeResult(
+                                            rootOperation.GetEnclosingControlFlowGraph(),
+                                            operationBlockAnalysisContext.Compilation,
+                                            operationBlockAnalysisContext.OwningSymbol,
+                                            operationBlockAnalysisContext.Options,
+                                            Rule,
+                                            sourceInfoSymbolMap,
+                                            taintedDataConfig.GetSanitizerSymbolMap(SinkKind),
+                                            sinkInfoSymbolMap,
+                                            operationBlockAnalysisContext.CancellationToken);
+
+                                        foreach (TaintedDataSourceSink sourceSink in taintedDataAnalysisResult.TaintedDataSourceSinks)
+                                        {
+                                            if (!sourceSink.SinkKinds.Contains(SinkKind))
+                                            {
+                                                continue;
+                                            }
+
+                                            foreach (SymbolAccess sourceOrigin in sourceSink.SourceOrigins)
+                                            {
+                                                Diagnostic diagnostic = Diagnostic.Create(
+                                                    Rule,
+                                                    sourceOrigin.Location,
+                                                    additionalLocations: new Location[] { sourceSink.Sink.Location },
+                                                    messageArgs: new object[] {
+                                                        sourceOrigin.Symbol.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat),
+                                                        "",
+                                                        sourceSink.Sink.Symbol.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat),
+                                                        ""});
+                                                operationBlockAnalysisContext.ReportDiagnostic(diagnostic);
```

Comment:
I included method names, so when people looked at bugs, they'd have an idea of where the source and sink are.